timeutil/human.go

```diff
@@ -0,0 +1,53 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package timeutil
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/snapcore/snapd/i18n"
+)
+
+func noon(t time.Time) time.Time {
+	y, m, d := t.Date()
+	return time.Date(y, m, d, 12, 0, 0, 0, t.Location())
+}
+
+// Human turns the time (which must be in the past) into something
+// more meant for human consumption.
+func Human(then time.Time) string {
+	return humanTimeSince(then, time.Now())
+}
+
+func humanTimeSince(then, now time.Time) string {
+	then = then.Local()
+	now = now.Local()
+
+	switch d := int(math.Ceil(noon(now).Sub(noon(then)).Hours() / 24)); d {
+	case 0:
+		return then.Format(i18n.G("today at 15:04-07"))
```

Comment:
I think we can drop the numerical timezone offset (`-07`) as I'd expect the human version to be in local time anyway. If any, perhaps `MST` would be better.

Reply:
I'd rather have it there because what's local to `snap` might not be local to the user (in case of remote access, but also in case of misconfiguration). I guess the three-letter one is usually more friendly.

Reply:
I think I tend to prefer the numeric one because I never know what offset PST is, for example, or if it's a stress-related mind bug. Also growing up in a place where the timezone was ARS doesn't help.